ambari-server/src/main/python/ambari_server/setupTrustedProxy.py

```diff
@@ -0,0 +1,154 @@
+
+
+def populate_tproxy_configuration_property(properties, tproxy_user_name, property_name, question_text_qualifier):
+  resolved_property_name = property_name.replace(PROXYUSER_PLACEHOLDER, tproxy_user_name)
+  resolved_property_value = get_value_from_dictionary(properties, resolved_property_name, WILDCARD_FOR_ALL)
+  resolved_property_value = get_validated_string_input("Allowed {0} for {1} ({2})? ".format(question_text_qualifier, tproxy_user_name, resolved_property_value), resolved_property_value, REGEX_ANYTHING, "Invalid input", False)
+  properties[resolved_property_name] = resolved_property_value
+
+
+def add_new_trusted_proxy_config(properties):
+  tproxy_user_name = get_validated_string_input("The proxy user's (local) username? ", None, REGEX_ANYTHING, "Invalid Trusted Proxy User Name", False, allowEmpty=False)
+  populate_tproxy_configuration_property(properties, tproxy_user_name, PROXYUSER_HOSTS, "hosts")
+  populate_tproxy_configuration_property(properties, tproxy_user_name, PROXYUSER_USERS, "users")
+  populate_tproxy_configuration_property(properties, tproxy_user_name, PROXYUSER_GROUPS, "groups")
+  return get_YN_input("Add another proxy user [y/n] (n)? ", False)
+
+
+def update_tproxy_conf(ambari_properties, tproxy_configuration_properties, admin_login, admin_password):
+  request_data = {
+    "Configuration": {
+      "category": "tproxy-configuration",
+      "properties": {
+      }
+    }
+  }
+  request_data['Configuration']['properties'] = tproxy_configuration_properties
+  perform_changes_via_rest_api(ambari_properties, admin_login, admin_password, TPROXY_CONFIG_API_ENTRYPOINT, 'PUT', request_data)
+
+
+def remove_tproxy_conf(ambari_properties, admin_login, admin_password):
+  perform_changes_via_rest_api(ambari_properties, admin_login, admin_password, TPROXY_CONFIG_API_ENTRYPOINT, 'DELETE')
+
+
+def validate_options(options):
+  errors = []
+  if options.tproxy_enabled and not re.match(REGEX_TRUE_FALSE, options.tproxy_enabled):
+    errors.append("--tproxy-enabled should be to either 'true' or 'false'")
+
+  if len(errors) > 0:
+    error_msg = "The following errors occurred while processing your request: {0}"
+    raise FatalException(1, error_msg.format(str(errors)))
+
+
+def setup_trusted_proxy(options):
+  print_info_msg("Setup Trusted Proxy")
+
+  server_status, pid = is_server_runing()
+  if not server_status:
+    err = 'Ambari Server is not running.'
+    raise FatalException(1, err)
+
+  if not get_silent():
+    validate_options(options)
+
+    ambari_properties = get_ambari_properties()
+
+    admin_login, admin_password = get_ambari_admin_username_password_pair(options)
+    properties = get_trusted_proxy_properties(ambari_properties, admin_login, admin_password)
+
+    if not options.tproxy_enabled:
+      tproxy_support_enabled = get_value_from_dictionary(properties, TPROXY_SUPPORT_ENABLED)
+
+      if tproxy_support_enabled:
+        if tproxy_support_enabled and 'true' == tproxy_support_enabled:
```

Comment:
why do we need to check the tproxy_support_enabled once again here?

Reply:
At this stage `tproxy_support_enabled` is a String and can be undefined.

Reply:
But we're checking in line 123 and in line 124 too.

```python
  if tproxy_support_enabled:
    if tproxy_support_enabled and ..:
```